helm/scylla/templates/serviceaccount.yaml

```diff
@@ -0,0 +1,16 @@
+{{- if (and .Values.serviceAccount.create .Values.serviceAccount.annotations) -}}
+# Even though Scylla Operator creates and manages Service Account used by ScyllaClusters
+# we have to keep it here to keep backward compatibility of setting custom annotations via helm chart.
+# Operator will adopt it anyways on creation and updates.
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "scylla.serviceAccountName" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+  {{- include "scylla.labels" . | nindent 4 }}
```

Comment:
this should be fixed to what the operator labels are and there should be a unit test to validate it's in sync. we want to drop the forced adoption eventually and adopt only those matching the selector

Reply:
Good point, synced labels with what is set on our end. 
Helm doesn't have unit test library, it only allows to spawn pods verifying deployment  - like checking whether proper auth is set to DB succeded. 
There're third-party plugins adding unit tests, although i haven't used any. 
Evaluation and integration seems out of scope of this PR, verified output manually.
Leaving evaluation of unit tests library as separate issue:   https://github.com/scylladb/scylla-operator/issues/938

Reply:
I guess QA is gonna the annotation path for helm, so this is fine to address later